Add more description for .option directive (#68)
riscv-non-isa/riscv-asm-manual: riscv-asm.md

.macro       | name arg1 [, argn]             | begin macro definition \argname to substitute
.endm        |                                | end macro definition
.type        | symbol, @function              | accepted for source compatibility

> .option      | {rvc,norvc,pic,nopic,relax,norelax,push,pop} | RISC-V options, more detailed description in [.option](#.option).

Reviewer: Reformat table, to realign the `|`s?

Nit: "RISC-V options. Refer to [.option](#.option) for a more detailed description."

Author: I would prefer don't realign this time since we might need to realign for `.option arch,` soon, my thought is we can did that when converting the format to AsciiDoc.

> .option      | {rvc,norvc,pic,nopic,relax,norelax,push,pop} | RISC-V options, more detailed description see [.optino](#.option).

Reviewer 2: ```suggestion
.option      | {rvc,norvc,pic,nopic,relax,norelax,push,pop} | RISC-V options, more detailed description in [.option](#.option).
```

.balign      | b,[pad_val=0]                  | byte align
.zero        | integer                        | zero bytes

### <a name=.option></a> `.option`

#### `rvc`/`norvc`

Enable/disable C-extension for following code region.

#### `pic`/`nopic`

> Set the code mode is PIC (position independent code) or non-PIC mode, this will
> affect how `la` expand.

Reviewer 3: ```suggestion
Set the code mode to PIC (position independent code) or non-PIC mode, this will
affect how `la` expands.
```

.balign      | b,[pad_val=0]                  | byte align
.zero        | integer                        | zero bytes

### <a name=.option></a> `.option`

#### `rvc`/`norvc`

> Enable/disable C-extension for following code region.

Reviewer 2: ```suggestion
Enable/disable C-extension for the following code region.
```

> Enable/disable C-extension for the following code region.

Reviewer: "C-extension" -> "The C extension"

I don't know if the word "region" in "the following code region" kind of implies that this has limited scope, like an `#if ... #endif` region. I'll let the native speakers nitpick that :)

Author: @aswaterman did you having some suggestion on the wording about `region`?

#### `pic`/`nopic`

> Set the code mode to PIC (position independent code) or non-PIC mode, this will

Reviewer 2: ```suggestion
Set the code model to PIC (position independent code) or non-PIC, this will
```

> affect how `la` expands.

Reviewer: Nit: "Set the code model to PIC (position independent code) or non-PIC. This will affect the expansion of the `la` pseudoinstruction."
Is there some place we can link to that details how `la` is expanded with the various models?

Author: Add reference to pseudoinstructions

#### `relax`/`norelax`

> Enable/disable linker relaxation for following code region if the linker

Reviewer 2: ```suggestion
Enable/disable linker relaxation for the following code region if the linker
```

> supports it.

Reviewer: "if the linker supports it" -> I think this enables it unconditionally, no? At least Clang will just enable it, and if the linker doesn't support it (LLD...) then it craps out.

I don't know if it's worth mentioning that this also includes emission of`R_RISCV_ALIGN` relocations?
